### ISSUE TITLE
[BUG] Can't easily delete a page that has versions

### ISSUE THREAD

**Issue by waveywhite** [django-cms/django-cms#8068]
## Description

It is impossible to delete a page from the CMS admin if it has versions, unless a setting is changed in settings.py.

## Steps to reproduce
1. Create a page
2. Edit the page and publish, creating at least one version
3. In CMS admin go to Page Contents
4. Attempt to delete the page in question
5. A message appears saying that the versions are protected and it's impossible to delete the page

## Expected behaviour

It is possible to delete a page using the user interface.

## Actual behaviour

You get a notice about protected objects and there's no way of overcoming this from the admin UI

## Additional information (CMS/Python/Django versions)

If you add the following line to settings.py then you can delete pages. Surely this should be the default behaviour?:

"""
DJANGOCMS_VERSIONING_ALLOW_DELETING_VERSIONS = True
"""

## Do you want to help fix this issue?

* [ ] Yes, I want to help fix this issue and I will join the channel #pr-reviews on [the Discord Server](https://discord-pr-review-channel.django-cms.org) to confirm with the community that a PR is welcome.
* [ x] No, I only want to report the issue.

**Comment by fsbraun** [comment]
@waveywhite Thanks for sharing this. It is the default behavior, since it preserves the full history of a website. 

I understand that this is surprising, and there have been discussions to remove the "Delete page..." entry from the toolbar menu (#7748). 

I have created a PR to the project template to include the `DJANGOCMS_VERSIONING_ALLOW_DELETING_VERSIONS` setting here https://github.com/django-cms/cms-template/pull/8

**Comment by fsbraun** [comment]
Yes, you can delete pages in the page tree and from the frontend in the page menu of the toolbar.

**Comment by waveywhite** [comment]
Sorry I'm a little confused. To be clear then, the discussion to remove the delete page action is just from the frontend UI and we'll still be able to delete a page using the CMS admin UI?

**Comment by fsbraun** [comment]
Yes, indeed. It is to avoid unwanted deletion of (child) pages. For now, the delete menu in the frontend stays. In both cases, the versioning setting needs to be active (if you use versioning).

**Comment by gagnesteph** [comment]
I think that the 1st page create ca ever be deleted execpt directly in the database, looks like a bug to me.

Am I wrong?

**Comment by fsbraun** [comment]
@gagnesteph You can delete all pages.
